### ISSUE TITLE
[Bug]: Logto does not work with Cloudflare Tunnel

### ISSUE THREAD

**Issue by clho40** [coollabsio/coolify#3653]
### Error Message and Logs

I have my domain name parked at Cloudflare Tunnel and have HTTPS redirect set up. I can browse the Coolify dashboard without any issue. I can also browse some of my Python applications when I set the domain name to `http://my-python-app.mydomain.com` without any problem, Cloudflare automatically redirects the traffic to `https`

However, when I set up a Logto application, and filled in the `ADMIN_ENDPOINT` url as `http://logto-admin.mydomain.com`, I get the following error when trying to create an account:
```
{
    "code": "oidc.invalid_redirect_uri",
    "message": "`redirect_uri` did not match any of the client's registered `redirect_uris`.",
    "error": "invalid_redirect_uri",
    "error_description": "redirect_uri did not match any of the client's registered redirect_uris",
    "state": "JKtuiY1FXwScYAPeMEGwBCFX80oyLtMx3C2oS9E-9ohfbboh9qorZCsSDzfkaGBb3bV4lAZIWUqmIecTtKdjHg",
    "iss": "http://logto-admin.mydomain.com/oidc"
}
```
I assume that's because Cloudflare redirected the traffic to `https` and it does not match the `ADMIN_ENDPOINT`

If I set the `ADMIN_ENDPOINT` to `https://logto-admin.mydomain.com` I get - **logto-admin.mydomain.com redirected you too many times.**

How do I solve this problem?

### Steps to Reproduce

1. Setup Cloudflare Tunnel for the Coolify server
2. Enable HTTPS redirect, SSL/TLS forwarding in Cloudflare
3. Install Logto
4. Set the `ADMIN_ENDPOINT` to `http://logto-admin.yourdomain.com`
5. Browse the URL in your browser
6. Click the "Create Account" button

### Coolify Version

v4.0.0-beta.345

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Ubuntu 24

**Comment by clho40** [comment]
![Screenshot 2024-10-01 093851](https://github.com/user-attachments/assets/f0949dd0-a244-462f-a29e-b0845b7fd14c)
![Screenshot 2024-10-01 093947](https://github.com/user-attachments/assets/291a03f1-a1f1-4a6e-ad41-54799fc044e1)

**Comment by djsisson** [comment]
Not all services work using http in a tunnel.

If there is no optional env for you to change (endpoint) then you would need to switch to strict/full and then pass through https on the tunnel

**Comment by clho40** [comment]
@djsisson  Hi, thanks for your response. I am using Full (Strict) mode on Cloudflare but still getting the same problem. Can you please elaborate?

**Comment by djsisson** [comment]
@clho40 if they dont give you the ability to set the redirecturis based on an env.

ps what does the base endpoint env do? or is that jsut the normal app?

can u check that first

**Comment by thereis** [comment]
Hello, I do have the same issue as him, just a cloudflare tunnel configured and no other changes done into the configuration file. When you enter in the admin panel, it returns the same error after clicking on "Create account"

**Comment by djsisson** [comment]
if you are unable to set the https redirect url, from any of the env, then you will need to set https pass through

for this you will need to create an origin certificate in cf
use domain and *.domain
install to /data/coolify/proxy/certs

thenb create a dynamic file and add the following

```yaml
tls:
  stores:
    default: {  }
  certificates:
    -
      certFile: /traefik/certs/domain.crt
      keyFile: /traefik/certs/domian.key
```

change names above as required

after saving check proxy logs to make sure you see no errors about invalid certificates

once that is done change cf ssl/tls to full/strict and https always
then change all of your domains in coolify to https ( you can jsut do this for logto only if you wish)
as well as change your tunnels , thats are currently set as http -> localhost 80
to now be https - localhost 443
finally for each of those tunnels goto additional
TLS - origin server name and enter your main domain name here

you will now have https all the way to your containers

**Comment by birdyboy18** [comment]
The above solution worked for me, the origin server name was the part that I was missing in my setup that got it all to work for me.

I've spent some time writing up a guide that's a more wordy version of what @djsisson has said 😂 . I've put a PR out for it!

Thanks for that final bit that got this working

https://github.com/coollabsio/documentation-coolify/pull/50

**Comment by peaklabs-dev** [comment]
I have updated the documentation with the details that will resolve this issue. It will be available soon.
